src/thermo/Phase.cpp

```diff
@@ -338,10 +370,18 @@ void Phase::setMassFractionsByName(const compositionMap& yMap)
     vector_fp mf(m_kk, 0.0);
     for (const auto& sp : yMap) {
         try {
-            mf[m_speciesIndices.at(toLowerCopy(sp.first))] = sp.second;
+            mf[m_speciesIndices.at(sp.first)] = sp.second;
         } catch (std::out_of_range&) {
-            throw CanteraError("Phase::setMassFractionsByName",
-                               "Unknown species '{}'", sp.first);
+            size_t loc = npos;
+            if (!m_caseSensitiveSpecies) {
+                loc = findSpeciesLower(sp.first);
```

Comment:
This should use `speciesIndex`, the same as in `setMoleFractionsByName`, right?

Reply:
Correct. I missed this one ... resulting in a tremendous simplification :grin: